snap/info.go

```diff
@@ -208,6 +208,17 @@ func ScopedSecurityTag(snapName, scopeName, suffix string) string {
 	return fmt.Sprintf("snap.%s.%s.%s", snapName, scopeName, suffix)
 }
 
+// SplitSecurityTag will extract the snap name from a string of the form
+// `snap.<snap>[.<app>]`. If the tag is not of that form, returns an error.
+func SplitSecurityTag(tag string) (string, error) {
+	errMsg := fmt.Errorf("invalid security tag: must be of the form 'snap.<snap>' or 'snap.<snap>.<app>': %q", tag)
+	l := strings.SplitN(tag, ".", 3)
+	if len(l) < 2 || l[0] != "snap" || len(l[1]) == 0 {
```

Comment:
It doesn't prevent us from getting the snap name but to comply with the form in the comment, this should also fail when `... || (len(l) == 3 && len(l[2]) == 0)`. Otherwise, `snap.foo.` would pass